pymeasure/instruments/rigol/Rigol_DP932U.py

```diff
@@ -0,0 +1,152 @@
+import logging
+from pymeasure.instruments import Instrument, SCPIMixin
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+# Set up logging
+logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s")
+
+
+def check_error_decorator(func):
+    """
+    Decorator to automatically check for errors after executing a method.
+    Logs errors and raises exceptions if the device reports an error.
+    """
+    def wrapper(self, *args, **kwargs):
+        result = func(self, *args, **kwargs)  # Call the wrapped method
+        error = self.check_error()  # Check for errors
+        if error and error != "No error":
+            logging.error(f"System Error after  {func.__name__}: {error}")
+            raise RuntimeError(f"System Error: {error}")
+        return result
+    return wrapper
+
+
+class RigolDP932U(SCPIMixin, Instrument):
+    """
+    PyMeasure interface for the Rigol DP932U DC Power Supply Unit.
+
+    This class provides methods to control the active channel,
+    voltage, current, output state, and connection mode of the device.
+    It also includes methods to measure voltage and current, reset the device,
+    and check for errors.
+    """
+
+    def __init__(self, adapter, name="Rigol DP932U Power Supply", **kwargs):
+        """
+        Initialize the Rigol DP932U DC Power Supply Unit.
+
+        :param adapter: The communication adapter (e.g., USB or GPIB) to connect to the instrument.
+        :param kwargs: Additional arguments for instrument initialization.
+        """
+
+        if not adapter:
+            raise ValueError("Adapter cannot be None. Provide a valid communication adapter.")
+        kwargs.pop("name", None)
+        super().__init__(adapter, name, **kwargs)
+
+    control_channel = Instrument.control(
```

Comment:
You do not have to add "Control" in front of a control, just name it "active_channel".